launcher.py

```diff
@@ -0,0 +1,405 @@
+        if system in self.check_cmd:
+            return self.check_cmd[system]
+        if "all" in self.check_cmd:
+            return self.check_cmd["all"]
+        return [f"{self.package_name.get(system, self.name)} --version"]
+
+    def _get_install_commands(self, system):
+        if system in self.install_cmd:
+            return self.install_cmd[system]
+        if "all" in self.install_cmd:
+            return self.install_cmd["all"]
+        if system not in self._installers:
+            print(f"Unspported sytem {system}. Please Install {self.name} manually.")
+            sys.exit(1)
+
+        return self._installers[system]()
+
+    def _get_install_commands_linux(self):
+        import distro
+
+        distro = distro.id().lower()
+        if distro not in self._linux_distro_map:
+            print(f"Unsupported linux distro {distro}. Please Install {self.name} manually.")
+            sys.exit(1)
+
+        return [f"{self._linux_distro_map[distro]} {{self.package_name.get(system, self.name)}}"]
+
+    def _get_install_commands_darwin(self):
+        return [f"brew install {{self.package_name.get(system, self.name)}}"]
+
+    def _get_install_commands_windows(self):
+        if not check_msys2_installed():
+            install_msys2()
+        return [f"pacman -S --needed --noconfirm {{self.package_name.get(system, self.name)}}"]
+
+
+# pkg-config should come before sndfie and nlohmann-json
+dependencies = [
+    DependencyHandler("cmake", {"Windows": "mingw-w64-x86_64-cmake"}, {"all": ["cmake --version"]}),
+    DependencyHandler(
+        "g++",
+        {"Windows": "mingw-w64-x86_64-gcc", "Darwin": "gcc"},
+        {"all": ["g++ --version"]},
+        {"Windows": ["pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain"]},
+    ),
+    DependencyHandler("pkg-config"),
+    DependencyHandler("ffmpeg", {"Windows": "mingw-w64-x86_64-ffmpeg"}, {"all": ["ffmpeg -version"]}),
+    DependencyHandler(
+        "libsndfile",
+        {"Windows": "mingw-w64-x86_64-libsndfile", "Linux": "libsndfile1-dev"},
+        {"all": ["pkg-config --exists sndfile"]},
+    ),
+    DependencyHandler(
+        "nlohmann-json",
+        {"Windows": "mingw-w64-x86_64-nlohmann-json", "Linux": "nlohmann-json3-dev"},
+        {"all": ["pkg-config --exists nlohmann_json"]},
+    ),
+]
+
+
+def check_msys2_installed():
+    try:
+        run_command("pacman --version")
+        return True
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        return False
+
+
+def install_msys2():
+    try:
+        global msys2_root_path
+        installer_url = (
+            "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.sfx.exe"
+        )
+        installer_name = "msys2-installer.exe"
+
+        msys2_root_path = "C:\\msys64"
+
+        print("Downloading MSYS2 installer...")
+        run_command(f"curl -L -o {installer_name} {installer_url}")
```

Comment:
Possible that the host may have connectivity issues.

Reply:
Seeing this again, we should actually check internet access at the start and prompt the user in case of an issue.

Reply:
I only able to correct the issues above in available time. I will fix this.